### DISCUSSION TITLE
Procedures for character consistency

### DISCUSSION THREAD

**Discussion by gabriel-oj** [lllyasviel/Fooocus#3491] [Q&A] [answered]
Guys,

I have about 50 full body images of a woman (real pictures, AI edited), and I'd like to generate AI images keeping a perfect character consistency for the full body and face, based on these images.

My first approach to achieve this was by training a LoRA using the 50 images as my dataset. Here is the dataset below, only for reference:

![a](https://github.com/user-attachments/assets/bb96e390-36ed-49fe-889a-e505e8cf952b)

My training variations, however, didn't result in good LoRAs as I wanted. Although they kept face and body consistency in some of the generated images, they had too much imperfections and deformities, and also lacked realism.

Since my GPU is not very good for training (RTX 4060 Ti 8GB) and each training takes a lot of time for me, I'm now thinking about alternative approaches to achieve my goal, so I won't waste any more time on these time-consuming training attempts.

Considering I do have good reference images, what sequence of procedures would you suggest to achieve this, efficiently?

I would prefer to not have to depend on a solution that involves a thorough inpaint work on each image, since this would be counterproductive, when thinking about generating many images each time.

Any help is much appreciated.

Thanks in advance.

Gabriel

**Comment by Yogesh-DevHub** [comment, marked as answer]
Hi, 
you said "a solution that involves a thorough in-paint work on each image" is not practical for multiple images. Actually, you can do it simultaneously while generating the image if you opt for enhance feature before generating the image.

Here is a video explaining the same scenario: https://youtu.be/180ovCHT-Ss?t=197&feature=shared

**Reply by gabriel-oj** [reply]
Yes, I heard this feature was introduced in a recent update. I'll look into it. Thanks!
